modern-publish-plugin.md

```diff
@@ -0,0 +1,222 @@
+```
+publishing {
+    publications {
+        create<MavenPublication>("maven") {
+            from(components["java"])
+        }
+    }
+}
+
+java {
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+The plugin doesn't need to offer anything for sources and javadocs jars. It will
+simply create a publication using `components["java"]`.
+
+#### `com.vanniktech.maven.publish.gradle`
+
+Meant to be used with `java-gradle-plugin`.
+
+Pure Gradle configuration:
+```
+java {
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+Gradle automatically adds a publication when both `java-gradle-plugin` and 
+`maven-publish` are applied. There is no need for our own APIs, so this plugin
+does not have any functionality of it's own.
+
+#### `com.vanniktech.maven.publish.android`
+
+Meant to be used with `com.android.library`.
+
+Pure Gradle configuration:
+```
+afterEvaluate {
+    publishing {
+        publications {
+            create<MavenPublication>("maven") {
+                from(components["release"])
+            }
+        }
+    }
+}
+```
+
+
+There is no API available to create source and javadoc jars for Android 
+projects. The plugin will provide an API to configure those as well as 
+which variant to publish. These properties/methods can be removed/deprecated once
+AGP offers it's own methods and adds multi variant publishing.
+
+```
+androidPublishCompat {
+  androidVariantToPublish = "release"
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+#### `com.vanniktech.maven.publish.kotlin.js`
+
+Meant to be used with `org.jetbrais.kotlin.js`.
+
+Pure Gradle configuration:
+```publishing {
+    publications {
+        create<MavenPublication>("maven") {
+            from(components["kotlin"])
+            artifact(project.tasks.named("kotlinSourcesJar"))
+        }
+    }
+}
+```
+
+
+There is no API available to create source and javadoc jars for Kotlin JS
+projects, but there is at least a task for sources. The plugin will provide 
+an API to configure those. These properties/methods can be removed/deprecated once
+Kotlin offers it's own methods and adds multi variant publishing.
+
+```
+androidPublishCompat {
+  androidVariantToPublish = "release"
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+
+#### `com.vanniktech.maven.publish.kotlin.mpp`
+
+Meant to be used with `org.jetbrais.kotlin.multiplatform`.
+
+Pure Gradle configuration: n/a
+
+Like `java-gradle-plugin` the Kotlin MPP plugin sets up a publication 
+automatically. It also adds source jars automatically. This plugin adds
+an API to add javadoc publishing. There still is a `withSourcesJar()` which will
+add empty source jars to all publications missing one, this can be removed once
+the behavior is fixed on the Kotlin side.
+
+```
+kotlinMultiplatformPublishCompat {
+  withSourcesJar()
+  withJavadocJar()
+}
+```
+
+#### `com.vanniktech.maven.publish`
+
+- calls `publishToMavenCentral()`
+- sets `groupId`, `version` and `baseArtifactId` based on Gradle properties
```

Comment:
I agree which is why we don't have anything for these in the new API see the comments here https://github.com/vanniktech/gradle-maven-publish-plugin/pull/173#discussion_r532041099. This line is just for the old (current) plugin to keep it's current behavior like it does for other properties. Eventually this can get deprecated as well. We already supported setting group and version on the project yourself for a while and with 0.14.0 we removed the requirement to set these 3 gradle properties.

Reply:
Sorry, I was reading the PR from mobile, I missed some things 😢